### PR TITLE
add recover middleware and some enhances

### PR DESCRIPTION
**Pull request by raghavharness**
Added recovery middleware for poller and remove nil pointer exception if hibernate start operation fails. If nil pointer comes, dlite restarts.

### CODE REVIEW

File: command/harness/dlite/handler.go [hunk 1]
```diff
@@ -70,3 +72,21 @@ func handleStatus(p *poller.Poller) http.HandlerFunc {
 		io.WriteString(w, enabledStatus) //nolint: errcheck
 	}
 }
+
+func pollerMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if r := recover(); r != nil {
+				// Handle the panic here, log it, or respond with an error message.
+				fmt.Println("Panic occurred:", r)
```
Comment (vistaarjuneja) [inline command/harness/dlite/handler.go:81]: Nit: Use logger instead of fmt

File: command/harness/dlite/handler.go [hunk 2]
```diff
@@ -70,3 +72,21 @@ func handleStatus(p *poller.Poller) http.HandlerFunc {
...
+				errMsg := "Unable to handle request, something went wrong."
```
Comment (vistaarjuneja) [inline command/harness/dlite/handler.go:82]: you could do err = `fmt.Errorf("http: panic: %v", r)`

File: command/harness/setup.go [hunk 3]
```diff
@@ -219,24 +219,28 @@ func handleSetup(
...
+			out, logErr := poolManager.InstanceLogs(context.Background(), pool, instanceID)
 			if logErr != nil {
 				logr.WithError(logErr).Errorln("failed to fetch console output logs")
 			} else {
 				// Serial console output is limited to 60000 characters since stackdriver only supports 64KB per log entry
 				l := math.Min(float64(len(out)), 60000) //nolint:gomnd
-				logrus.WithField("id", instance.ID).
-					WithField("instance_name", instance.Name).
-					WithField("ip", instance.Address).
+				logrus.WithField("id", instanceID).
```
Comment (harshjain12) [inline command/harness/setup.go:235]: can we add account id ?

**Review by harshjain12 — APPROVED** [review]

File: command/harness/setup.go (hunk 3 above)
Comment (raghavharness) [reply command/harness/setup.go:235]: Will change from logrus to logr in next pr so that it takes values from context

**Pull request merged**